### PR TITLE
LUN-86: Reuse code for single choice exercises

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

File: src/screens/choice-exercises/services/ArticleChoiceExerciseService.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { SingleChoiceListItemType } from '../components/SingleChoiceListItem'
```
Comment (SteffiStuffel) [inline src/screens/choice-exercises/services/ArticleChoiceExerciseService.ts:1]: Almost same code that was previously in ArticleChoiceExerciseScreen

File: src/screens/choice-exercises/services/WordChoiceExerciseService.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { SingleChoiceListItemType } from '../components/SingleChoiceListItem'
```
Comment (SteffiStuffel) [inline src/screens/choice-exercises/services/WordChoiceExerciseService.ts:1]: Almost same code that was previously in SingleChoiceExerciseScreen

File: src/screens/choice-exercises/ChoiceExercisesScreen.tsx [hunk 3]
```diff
@@ -50,43 +57,31 @@ const LearnArticlesExerciseScreen = ({ navigation, route }: LearnArticlesExercis
...
-    setAnswerOptions(answerOptions)
-  }, [currentWord, documents])
-
   useEffect(() => {
-    buildAnswerOption()
-  }, [documents, currentWord, buildAnswerOption])
+    setAnswerOptions(service.buildAnswerOptions(documents, currentWord, count))
+  }, [documents, currentWord, service, count])
+
+  function isAnswerEqual(answer1: Answer, answer2: SingleChoiceListItemType): boolean {
```
Comment (klinzo) [inline src/screens/choice-exercises/ChoiceExercisesScreen.tsx:64]: Convert to arrow function

**Review by klinzo — APPROVED** [review]
I'd rather not have a `choice-exercises` sub dir in `screens` and would keep all components in the main components folder.

File: src/screens/choice-exercises/ChoiceExercisesScreen.tsx (hunk 3 above)
Comment (SteffiStuffel) [reply src/screens/choice-exercises/ChoiceExercisesScreen.tsx:64]: done.

**Comment by SteffiStuffel** [comment]
> I'd rather not have a `choice-exercises` sub dir in `screens` and would keep all components in the main components folder.

I thought you would, but I do not agree that this is good structuring. I won't find anything in the folder structur and also would have to you Intellij for this. For me this is a sign for a bad folder structure and I would like to try a more common and recommended way of structuring the files.

**Comment by klinzo** [comment]
Your choice, but imo you just go to the components folder and look for choice and find instantly what you are looking for? :D

**Pull request merged**